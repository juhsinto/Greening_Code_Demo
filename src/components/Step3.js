import React from "react";

import {useForm} from "react-hook-form";
import { useDispatch, useSelector } from "react-redux";
import { useHistory } from "react-router-dom";
import { updatePage3, updateFormProgress } from "../rootSlice";
import {Result} from "./Result"
import {Header} from "./Header"

import {Jumbotron,Container,Form,Button} from 'react-bootstrap';


export const Step3 = () => {

    const dispatch = useDispatch()
    const history = useHistory()
    const project_information = useSelector(state => state.page3.project_information)
    const project_comments = useSelector(state => state.page3.project_comments)
    
    
    const {register, handleSubmit} = useForm({defaultValues: {project_information, project_comments}});

    const onSubmit = (data) => {
        
        dispatch(
            updatePage3({
              ...data,
            })
          );
        dispatch(updateFormProgress(100));
        history.push("./step4")
    }

    const goBack = () => {
        dispatch(updateFormProgress(0));
        history.push("./step2");
    };

    return (
        <Container className="p-3">
            <Jumbotron>
                <Header />
                <form onSubmit={handleSubmit(onSubmit)}>
                    <div>
                        <label htmlFor="page1">Page 3 Details:</label>
                        <br />
                        <Form.Group>
                            <Form.Label>Project Information</Form.Label>
                            <Form.Control as="textarea" name="project_information" id="project_information"  ref={register} rows={3} />
                        </Form.Group>
                        <br />
                        <Form.Group>
                            <Form.Label>Project Comments</Form.Label>
                            <Form.Control as="textarea" name="project_comments" id="project_comments"  ref={register} rows={3} />
                        </Form.Group>
                        <Button onClick={goBack}>Previous</Button>{' '}
                        <Button type="submit">Next</Button>
                    </div>
                </form>
            </Jumbotron>
            <Result />
        </Container>
    )
}

export default Step3;
