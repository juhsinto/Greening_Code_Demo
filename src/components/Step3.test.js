import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import Step3 from './Step3';
import { store } from '../store';

it('renders Step3 correctly', () => {
  const tree = renderer.create(<Provider store={store}><Step3 /></Provider>).toJSON();
  expect(tree).toMatchSnapshot();
});