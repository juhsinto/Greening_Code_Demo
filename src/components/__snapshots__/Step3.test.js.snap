// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Step3 correctly 1`] = `
<div
  className="p-3 container"
>
  <div
    className="jumbotron"
  >
    <h1
      className="header"
    >
      A simple Questionnaire
    </h1>
    <form
      onSubmit={[Function]}
    >
      <div>
        <label
          htmlFor="page1"
        >
          Page 3 Details:
        </label>
        <br />
        <div
          className="form-group"
        >
          <label
            className="form-label"
          >
            Project Information
          </label>
          <textarea
            className="form-control"
            id="project_information"
            name="project_information"
            rows={3}
          />
        </div>
        <br />
        <div
          className="form-group"
        >
          <label
            className="form-label"
          >
            Project Comments
          </label>
          <textarea
            className="form-control"
            id="project_comments"
            name="project_comments"
            rows={3}
          />
        </div>
        <button
          className="btn btn-primary"
          disabled={false}
          onClick={[Function]}
          type="button"
        >
          Previous
        </button>
         
        <button
          className="btn btn-primary"
          disabled={false}
          type="submit"
        >
          Next
        </button>
      </div>
    </form>
  </div>
  <p>
    JSON data:
  </p>
  <pre>
    {
  "page1": {},
  "page2": {},
  "page3": {},
  "formProgress": 0
}
  </pre>
  <p>
    Form Progress:
  </p>
  <div
    className="progress"
  >
    <div
      aria-valuemax={100}
      aria-valuemin={0}
      aria-valuenow={0}
      className="progress-bar"
      role="progressbar"
      style={
        Object {
          "width": "0%",
        }
      }
    >
      0%
    </div>
  </div>
</div>
`;
